packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -428,8 +435,12 @@ const DataGridProVirtualScroller = React.forwardRef<
         {rightRenderContext && (
           <VirtualScrollerPinnedColumns
             ref={rightColumns}
-            ownerState={{ side: GridPinnedPosition.right }}
-            className={classes.rightPinnedColumns}
+            ownerState={{
+              side: theme.direction === 'ltr' ? GridPinnedPosition.right : GridPinnedPosition.left,
+            }}
+            className={
+              theme.direction === 'ltr' ? classes.rightPinnedColumns : classes.leftPinnedColumns
```

Comment:
A note: MUI Core mostly uses the opposite asserting:

```suggestion
              theme.direction === 'rtl' ? classes.leftPinnedColumns : classes.rightPinnedColumns;
```

If there is no strong argument for one vs. the other, I think that we should change it.